rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -366,6 +387,29 @@ module Impl {
     override string getAPrimaryQlClass() { result = "VariableAccess" }
   }
 
+  /** Holds if `e` occurs in the LHS of a (compound) assignment. */
+  private predicate assignLhs(Expr e) {
+    exists(BinaryExpr be |
+      be.getOperatorName().regexpMatch(".*=") and
```

Comment:
I don't feel strongly about this one, but I believe `matches` is still preferred over `regexpMatch` where possible???
```suggestion
      be.getOperatorName().matches("%=") and
```

Reply:
Yep, there was a discussion a long while back about whether the optimiser is more aware of `matches`, I can't remember the answer.  Anyway, up to you.